autotest/ogr/ogr_flatgeobuf.py

```diff
@@ -697,3 +697,22 @@ def test_ogr_flatgeobuf_column_metadata():
 
     ogr.GetDriverByName('FlatGeobuf').DeleteDataSource('/vsimem/test.fgb')
     assert not gdal.VSIStatL('/vsimem/test.fgb')
+
+def test_ogr_flatgeobuf_editing():
+    ds = ogr.GetDriverByName('FlatGeobuf').CreateDataSource('/vsimem/test.fgb')
+    lyr = ds.CreateLayer('test', geom_type = ogr.wkbPoint)
+    f = ogr.Feature(lyr.GetLayerDefn())
+    f.SetGeometry(ogr.CreateGeometryFromWkt('POINT (0 0)'))
+    lyr.CreateFeature(f)
+
+    ds = None
+
+    ds = ogr.Open('/vsimem/test.fgb', update=1)
+    lyr = ds.GetLayer(0)
+
+    assert lyr.TestCapability(ogr.OLCDeleteFeature) == 1
+    assert lyr.DeleteFeature(0) == 0
+    assert lyr.DeleteFeature(0) == ogr.OGRERR_NON_EXISTING_FEATURE
+    lyr.ResetReading()
+    f = lyr.GetNextFeature()
+    assert f is None
```

Comment:
test a bit light.
dataset should be reopended after editing to check eveything is fine. you'd probably also want to test SetFeature(), and all methods of OGRFlatGeobufEditableLayer

cleanup of /vsimem/test.fgb should be done, but probably something for the tear-down code

Reply:
agreed, I have done some expanded local tests indicating that the implementation has bugs. will be back with something better, I hope.

Reply:
Now extended quite a bit. I cannot verify that ReorderFields works and when investigating that I couldn't make it work on the csv driver either.